RecipeMatch/Views/DraggableViewBackground.m

```diff
@@ -172,35 +208,37 @@ - (void)cardSwipedRight:(UIView *)cardSwiped{
         cardsLoadedIndex++; // loaded a card, so have to increment count
         [self insertSubview:[loadedCards objectAtIndex:(MAX_BUFFER_SIZE-1)] belowSubview:[loadedCards objectAtIndex:(MAX_BUFFER_SIZE-2)]];
     }
-    DraggableView *nextCard = [loadedCards objectAtIndex:0];
-    [self updateHeartBtn:nextCard];
-    [self updateLikeCount:nextCard];
+    [self updateValues];
 }
 
 // called when like button is tapped
 - (void)tapLike:(id)sender{
     DraggableView *card = [loadedCards firstObject];
     NSString *shortId = [(NSString *)card.recipeId substringFromIndex:ID_INDEX];
-    
     [delegate checkLikeStatusFromDraggableViewBackground:card withCompletion:^(BOOL liked, NSError * _Nullable error){
         if(liked){
             [self.delegate unlikeRecipeFromDraggableViewBackgroundWithId:shortId andCompletion:^(BOOL succeeded, NSError * _Nonnull error) {
                 if(succeeded){
-                    [sender setImage:[UIImage imageNamed:@"heart-btn"] forState:UIControlStateNormal];
-                    [self updateLikeCount:card];
+                    [sender setImage:[UIImage imageNamed:HEART_IMG] forState:UIControlStateNormal];
+                    [self updateLikeCount];
                 }
             }];
         } else{
             [self.delegate postLikedRecipeFromDraggableViewBackgroundWithId:shortId recipeTitle:card.title.text image:card.imageUrl andCompletion:^(BOOL succeeded, NSError * _Nonnull error){
                 if(succeeded){
-                    [sender setImage:[UIImage imageNamed:@"heart-btn-filled"] forState:UIControlStateNormal];
-                    [self updateLikeCount:card];
+                    [sender setImage:[UIImage imageNamed:HEART_FILL_IMG] forState:UIControlStateNormal];
+                    [self updateLikeCount];
                 }
             }];
         }
     }];
 }
 
+- (void)detailsAction{
+    DraggableView *card = [loadedCards firstObject];
+    [(StreamViewController *)delegate showDetails:card];
```

Comment:
no need to cast delegate

Reply:
also, i noticed you haven't marked this (and the other) protocol method as `@required`, so it means it doesn't necessary means it will be implemented, and if you call it right away it will crash.
so you either declare it as `@required` or check that your delegate respond to a selector:
``` 
    if ([self.delegate respondsToSelector:@selector(showDetails:)]) {
        [self.delegate showDetails:card];
    }
 ```